pkg/client/round_trippper.go

```diff
@@ -0,0 +1,32 @@
+package client
+
+import (
+	"fmt"
+	"net/http"
+)
+
+type customHeadersRoundTripper struct {
+	header                http.Header
+	delegatedRoundTripper http.RoundTripper
+}
+
+func NewCustomHeadersRoundTripper(header http.Header, rt http.RoundTripper) http.RoundTripper {
+	return &customHeadersRoundTripper{
+		header:                header,
+		delegatedRoundTripper: rt,
+	}
+}
+
+var _ http.RoundTripper = &customHeadersRoundTripper{}
+
+func (c *customHeadersRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	for key, values := range c.header {
+		if req.Header.Get(key) != "" {
```

Comment:
when will this happen in our case? Is that the env has changed?

Reply:
This will happen when the custom header is trying to override a other header, for example `HOST`, or `Content-Type`